module/VuFind/src/VuFind/OAI/Server.php

```diff
@@ -593,33 +604,39 @@ protected function listRecords($verb = 'ListRecords')
             }
         }
 
-        // Figure out how many Solr records we need to display (and where to start):
-        if ($currentCursor >= $deletedCount) {
-            $solrOffset = $currentCursor - $deletedCount;
-        } else {
-            $solrOffset = 0;
-        }
-        $solrLimit = ($params['cursor'] + $this->pageSize) - $currentCursor;
+        // Figure out how many non-deleted records we need to display:
+        $recordLimit = ($params['cursor'] + $this->pageSize) - $currentCursor;
+        $cursorMark = $params['cursorMark'] ?? '';
 
         // Get non-deleted records from the Solr index:
         $set = $params['set'] ?? '';
         $result = $this->listRecordsGetNonDeleted(
-            $from, $until, $solrOffset, $solrLimit, $set
+            $from,
+            $until,
+            $cursorMark,
+            $recordLimit,
+            $set
         );
         $nonDeletedCount = $result->getResultTotal();
         $format = $params['metadataPrefix'];
         foreach ($result->getResults() as $doc) {
             if (!$this->attachNonDeleted($xml, $doc, $format, $headersOnly, $set)) {
-                $this->unexpectedError('Cannot load document');
+                // Record cannot be returned in the given format
+                continue;
```

Comment:
If we continue here, is the fact that we're not incrementing $currentCursor as a result going to throw anything off? Seems like we might get duplicate or unexpected results....